### PR TITLE
Increased Truncation limits

### PR DESCRIPTION
**Pull request by Raemon**

### CODE REVIEW

File: packages/lesswrong/components/comments/CommentsItem/CommentsItem.jsx [hunk 1]
```diff
@@ -98,8 +98,15 @@ class CommentsItem extends Component {
...
-    const { comment } = this.props
-    return comment.baseScore > 20 ? 1000 : 300
+    const { comment, currentUser, postPage } = this.props
+    if (currentUser.noCollapseCommentsPosts && postPage) {
+      return 10000000
+    }
+    if (currentUser.noCollapseCommentsFrontpage && !postPage) {
+      return 10000000
+    }
+    const commentIsRecent = comment.postedAt > new Date(new Date().getTime()-(2*24*60*60*1000));
```
Comment (jimrandomh) [inline packages/lesswrong/components/comments/CommentsItem/CommentsItem.jsx]: Add comment:
```suggestion
    const commentIsRecent = comment.postedAt > new Date(new Date().getTime()-(2*24*60*60*1000)); // past 2 days
```

File: packages/lesswrong/lib/editor/ellipsize.jsx [hunk 2]
```diff
@@ -37,7 +37,7 @@ export const commentExcerptFromHTML = (html, truncationCharCount) => {
   const htmlRemovedStyles = html.replace(/<style[\s\S]*?<\/style>/g, '');
 
   return Utils.sanitize(truncatise(htmlRemovedStyles, {
-    TruncateLength: truncationCharCount || excerptMaxChars,
+    TruncateLength: truncationCharCount - truncationCharCount/4 || excerptMaxChars,
```
Comment (jimrandomh) [inline packages/lesswrong/lib/editor/ellipsize.jsx]: Parenthesize, and also Math.floor this (truncationCharCount isn't guaranteed to be divisible by 4, although it happens to be with the parameters we've set.)

File: packages/lesswrong/lib/collections/users/custom_fields.js [hunk 3]
```diff
@@ -784,7 +784,37 @@ Users.addField([
...
+    fieldSchema: {
+      order: 70,
+      type: Boolean,
+      optional: true,
+      defaultValue: false,
+      canRead: ['guests'],
+      canUpdate: [Users.owns, 'sunshineRegiment', 'admins'],
+      canCreate: ['members'],
+      control: 'checkbox',
+      label: "Do not collapse comments (on Frontpage)"
```
Comment (jimrandomh) [inline packages/lesswrong/lib/collections/users/custom_fields.js]: This isn't actually about capital-f Frontpage; for the sake of consistent terminology, it should be "front page" (lower case and with a space).

Comment (jimrandomh) [inline packages/lesswrong/lib/collections/users/custom_fields.js]:
```suggestion
      label: "Do not collapse comments (on front page)"
```

File: packages/lesswrong/lib/editor/ellipsize.jsx (hunk 2 above)
Comment (jimrandomh) [reply packages/lesswrong/lib/editor/ellipsize.jsx]:
```suggestion
    TruncateLength: Math.floor(truncationCharCount - truncationCharCount/4) || excerptMaxChars,
```

File: packages/lesswrong/lib/collections/users/custom_fields.js (hunk 3 above)
Comment (Raemon) [inline packages/lesswrong/lib/collections/users/custom_fields.js]:
```suggestion
      label: "Do not collapse comments (on frontpage)"
```

File: packages/lesswrong/lib/collections/users/custom_fields.js [hunk 4]
```diff
@@ -784,7 +784,37 @@ Users.addField([
...
+    fieldSchema: {
+      order: 70,
+      type: Boolean,
+      optional: true,
+      defaultValue: false,
+      canRead: ['guests'],
+      canUpdate: [Users.owns, 'sunshineRegiment', 'admins'],
+      canCreate: ['members'],
+      control: 'checkbox',
+      label: "Do not collapse comments (on frontpage)"
```
Comment (Raemon) [inline packages/lesswrong/lib/collections/users/custom_fields.js]:
```suggestion
      label: "Do not collapse comments (on home page)"
```

**Pull request merged**